microsoft/fhir-server
# Adds reindex test

## Description

Refactors reindex integration tests and adds a new test to cover the scenario where a search parameter is created before the resources it impacts (more details can be found in [AB#79555](https://microsofthealth.visualstudio.com/Health/_workitems/edit/79555) which was fixed in #1650).

If this looks familiar: most of this refactoring already lives in the feature branch for SQL reindexing and was reviewed in #1577.

## Related issues
Addresses [AB#79650](https://microsofthealth.visualstudio.com/Health/_workitems/edit/79650).

## Testing
Manual testing:
- Ensured that the tests continue to pass for Cosmos DB
- Temporarily enabled the reindex tests for SQL to ensure that the tests run without fixture-related issues (although some tests fail, as expected)

## FHIR Team Checklist
- ✅ **Update the title** of the PR to be succinct and less than 50 characters
- ✅ **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- ✅ Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- ✅ Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)
